app/javascript/packs/onboardingRedirectCheck.jsx

```diff
@@ -42,7 +51,10 @@ document.ready.then(
       window.currentUser = currentUser;
       window.csrfToken = csrfToken;
 
-      if (redirectableLocation() && onboardCreator(currentUser)) {
+      if (
+        redirectableCreatorOnboardingLocation() &&
+        onboardCreator(currentUser)
+      ) {
         window.location = `${window.location.origin}/admin/creator_settings/new?referrer=${window.location}`;
       } else if (redirectableLocation() && !onboardingSkippable(currentUser)) {
         window.location = `${window.location.origin}/onboarding?referrer=${window.location}`;
```

Comment:
This should fix the redirect issue that @aitchiss and myself are having.

```suggestion
      if (onboardCreator(currentUser)) {
        if (redirectableCreatorOnboardingLocation()) {
          window.location = `${window.location.origin}/admin/creator_settings/new?referrer=${window.location}`;
        }
      } else if (redirectableLocation() && !onboardingSkippable(currentUser)) {
        window.location = `${window.location.origin}/onboarding?referrer=${window.location}`;
```

Reply:
Thank you so, so much for the help, @nickytonline -- I was unable to recreate the unexpected redirect to onboarding on my end, which made this difficult to debug. That said, this change still works as expected for me! 

I am super curious though: why does this fix the issue you both were experiencing?

Reply:
Try clearing the application state in the browser. There may be lingering user data in there.

![CleanShot 2021-12-03 at 12 25 53](https://user-images.githubusercontent.com/833231/144646166-f0200691-0900-4dd0-b7f4-4c76ebe8ba5b.png)